File: sentinel-backend/tests/archived-results/mixture-analyzer/analysis.js
const http = require('http');
const fs = require('fs');

function handleRequest(req, res) {
    const input = req.url.substring(1);

    const db_query = `SELECT * FROM users WHERE id=${input}`;

    require('child_process').exec(`ls /home/${input}`);

    fs.readFile('/data/' + input, (err, data) => {
        if (!err) res.end(data);
    });

    eval(input);

    const template = `<%= ${input} %>`;

    const ldap_filter = `(uid=${input})`;

    fs.readFile('/data/config.xml', (err, data) => {
        if (!err) res.end(data);
    });

    try {
        const deserialized_data = require('pickle').loads(input);
    } catch (e) { }

    res.setHeader("Set-Cookie", `session=${input}; HttpOnly=false`);

    const oauthUrl = `/oauth/callback?state=${input}`;
}

http.createServer(handleRequest).listen(8080);

File: sentinel-backend/tests/archived-results/mixture-analyzer/analysis.cpp
#include <iostream>
#include <string>
#include <fstream>

int main() {
    std::string input;
    getline(std::cin, input);

    const char* db_query = "SELECT * FROM users WHERE id=" + input;

    std::system(("ls /home/" + input).c_str());

    std::ifstream file("/data" + input);
    if (file.is_open()) {
        std::string content((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
        file.close();
    }

    const char* raw_query = "queryRawUnsafe(" + input + ")";

    const char* mongo_query = "$where=" + input;

    const char* ldap_filter = "(uid=" + input + ")";

    std::ifstream xml_file("/data/config.xml");
    if (xml_file.is_open()) {
        std::string content((std::istreambuf_iterator<char>(xml_file)), std::istreambuf_iterator<char>());
        xml_file.close();
    }

    const char* deserialized_data = input.c_str();

    system("kubectl run --image=privileged:image");

    system("docker run --network host -it bash");

    return 0;
}
